### ISSUE TITLE
Agent get unresponsive on slow connection after a while

### ISSUE THREAD

**Issue by Allowee** [Ylianst/MeshCentral#2066]
I have a machine on a 4G modem, located in a concrete room with minimal 4G coverage.
The problem I see is that the machine will work fine for some time and end up unresponsive.
After some time, like 1 hour (not using the agent), the connection becomes slow, really slow.. even a 'help' command in the console will take second and after that takes minutes.
Than it will disconnect after a few minutes (agent is offline message) and never come back online.
This is interesting as I would expect the websocket to be unresponsive and the Mesh Agent to reconnect.
I then use another remote tool and restart the Mesh Agent and all is back up and running smooth, meaning the 4G connection is fine.

Any tips to improve or debug this?

**Comment by Allowee** [comment]
Just set agentpong to see if it helps

**Comment by krayon007** [comment]
What OS is this agent running on?

**Comment by krayon007** [comment]
Also, what sleep state is the machine in? Is it set to always be awake, or is it set to idle to sleep?

**Comment by Ylianst** [comment]
FYI. If your bandwidth sensitive, you may want to try these lines in the settings section of config.json:

```
    "Compression": true,
    "WsCompression": true,
    "AgentWsCompression": true
```

This will enable websocket compression and you can expect a 5% to 15% reduction in traffic for remote desktop, terminal and files. The 3rd line is especially important since it enables agent to server compression.

In addition, in the domain section of the config.json, add:

```
    "minify": true
```

This will reduce the size of web pages.

**Comment by Ylianst** [comment]
What version of the MeshCentral server are you using?

**Comment by Allowee** [comment]
Windows Server 2019 Essentials - 1809 [17763]
Mesh: 0.7.14

Power plan high performance and never go to sleep.
When the agent was connected to the lan everything was fine.
Unresponsive times are also different each time, was unable to get a response from the agent in 10 minutes.

Side note: I have multiple clients on 4g and even a GPRS modem without problems. This site has the weakest signal.
Could be switching towers and losing the session or something

**Comment by Allowee** [comment]
added the options... running fine now for more than a hour now.
Will report back tomorrow morning

**Comment by Allowee** [comment]
I believe the added AgentPong fixed the issue. Will keep testing with the Agent in commandline mode to see when it re-connects.

Is there no ping-ponging enabled by default?

**Comment by krayon007** [comment]
By default the agent sends a ping after every 2 minutes of inactivity, using a websocket control frame.

Some reverse proxies don't correctly forward websocket control frames, which is why we also have an optional ping pong using a data frame.

**Comment by yomono** [comment]
> A bit unrelated, but if your bandwidth sensitive, you may want to try these lines in the settings section of config.json:
> 
> ```
>     "Compression": true,
>     "WsCompression": true,
>     "AgentWsCompression": true
> ```
> 
> This will enable websocket compression and you can expect a 5% to 15% reduction in traffic for remote desktop, terminal and files. The 3rd line is especially important since it enables agent to server compression.
> 
> In addition, in the domain section of the config.json, add:
> 
> ```
>     "minify": true
> ```
> 
> This will reduce the size of web pages.

Hi, sorry for ask this in another's user Issue but is there any downsides with this? Any reason should I NOT enable this options? Why are not default?

**Comment by Allowee** [comment]
Update:

I have the agent connected in the cli with "connect" to see when there is a reconnect.
This should have been at 3am when the 4G modem reconnects.
Now I just noticed the desktop, terminal and files tab are missing.
Very strange.

I will try to find a way to reporduce and file a new issue when I find something

**Comment by Allowee** [comment]
Now having any problems now for a few days.
Closing

**Comment by Ylianst** [comment]
FYI. Just published MeshCentral v0.7.29 with all new agents that may fix issues where agent slows down.
